core/client/fs/src/main/java/alluxio/client/file/options/InStreamOptions.java

```diff
@@ -74,8 +91,11 @@ public InStreamOptions(URIStatus status, OpenFilePOptions options,
 
     mStatus = status;
     mProtoOptions = openOptions;
-    mUfsReadLocationPolicy = BlockLocationPolicy.Factory.create(
-        alluxioConf.getClass(PropertyKey.USER_UFS_BLOCK_READ_LOCATION_POLICY), alluxioConf);
+    mUfsReadLocationPolicy = context.getReadBlockLocationPolicy(alluxioConf);
+    if (mUfsReadLocationPolicy == null) {
+      mUfsReadLocationPolicy = BlockLocationPolicy.Factory.create(
+          alluxioConf.getClass(PropertyKey.USER_UFS_BLOCK_READ_LOCATION_POLICY), alluxioConf);
```

Comment:
Can `mUfsReadLocationPolicy` be null?  I think this check is redundant.

Reply:
It can be null for ut, some ut mock the `FileSystemContext`